device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubSendTask.java

```diff
@@ -42,11 +46,11 @@ public void run()
         {
             synchronized (this.sendThreadLock)
             {
-                if (!this.transport.hasMessagesToSend() && !this.transport.hasCallbacksToExecute() && !this.transport.isClosed())
+                while (!this.transport.hasMessagesToSend() && !this.transport.hasCallbacksToExecute() && !this.transport.isClosed())
```

Comment:
Changing from if (...) to while (...) so that this thread can continue to wait if no work was available after it was woken up. This thread may be woken up by a message being queued, the periodic failsafe wakeup, or by spurious wakeups. In the latter two cases, the thread will usually just go back to sleep since no work is available to process.

For an explanation on spurious wakeups, see [this post](https://stackoverflow.com/questions/13148001/spurious-wakeups-wait-and-notifyall)